vtex-apps/shelf
# Use ProductListContext to send impression events (Fixing product impression issues)

#### What is the purpose of this pull request?
Use the ProductListContext to handle the product impression events, which solves the issue of products that were not seen yet being impressed.

Needs https://github.com/vtex-apps/product-list-context/pull/3

#### How should this be manually tested?

[Workspace](https://iaronaraujo--storecomponents.myvtex.com/)
then on console type pixelManagerEvents to check the events that are being sent when you see the shelf.

#### Types of changes
